### DISCUSSION TITLE
Export with zero value when delta-preferred counter is adopted

### DISCUSSION THREAD

**Discussion by aaron-ai** [open-telemetry/opentelemetry-java#4779] [Q&A]
I saw that if delta-preferred aggregation temporality is adopted, counter will export the delta value to the OpenTelemetry collector, but if there is no recoding during the current period, no value will be exported, actually 0 is what I expect.

How could I solve this? Thanks!

**Comment by jack-berg** [comment]
The behavior is intentional. In fact, reduced network egress is part of the appeal of delta based metric systems. 

There is no simple way to configure the SDK to achieve the behavior you're describing. However, if you really want that behavior and are willing to write some code, you could write a metrics exporter that records all metric streams it sees, and if it receives a batch without an entry it previously saw, fills in the gaps.
